sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.java

```diff
@@ -176,8 +176,7 @@ public void initBatch(
         // Initialize the missing columns once.
         if (colId == -1) {
           OnHeapColumnVector missingCol = new OnHeapColumnVector(capacity, dt);
-          missingCol.putNulls(0, capacity);
-          missingCol.setIsConstant();
+          missingCol.setAllNull();
```

Comment:
Oh, almost forgot to ask. 

Is `missingCol.setAllNull();` equivalent to 
```
missingCol.putNulls(0, capacity);
missingCol.setIsConstant();
```
? I thought `setAllNull` was an alias for the two method calls but it does not appear to be the case.

Reply:
Not quite the same. `setIsConstant` requires to allocate space for the constant elements, but `setAllNull` doesn't. This is useful for those missing columns that are of nested types.

Reply:
Is this related this PR or new test coverage?
This change looks like relying on the existing ORC tests only.
In that case, shall we make a separate PR for this ORC change?